### PR TITLE
Add readme

### PR DESCRIPTION
**Pull request by fbouliane**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+The project make sure that the requester for a lock will have it according to the order it asked for access. It also 
+adds the ability to extends a lock without having to release the lock.
+
+Why would I use this ?
+----------------------
+The normal behavior of redlock-py is to `return false` on a existing lock. The mainstream approach is to 
+poll the lock until acquisition. This process has two downsides :
+
+1. It doesn't warranty that the first requester will be the first to get the lock.
```
Comment (godp1301) [inline README.md]: IMO warranty is not a verb so it can't be used likewise. Perhaps guarantee would be better...

**Comment by JoProvost** [comment]
Looks good to me :+1:

File: README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+[![Build Status](https://travis-ci.org/internap/netman.svg?branch=master)](https://travis-ci.org/internap/redlock-fifo)
+[![PyPI version](https://badge.fury.io/py/redlock-fifo.svg)](http://badge.fury.io/py/redlock-fifo)
+
+Redlock-Fifo
+============
+
+Redlock-Fifo makes [Redlock-py](https://github.com/SPSCommerce/redlock-py) usable in a multi-threaded environment.
+
+This project makes sure that the requester will acquire a lock according to the order it is asked for. It also 
+adds the ability to extends a lock without having to release the lock.
```
Comment (idjaw) [inline README.md]: s/extends/extend/

File: README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+----------------------
+The normal behavior of redlock-py is to `return false` on an existing lock. The mainstream approach is to 
+poll the lock until acquisition. This process has two downsides:
+
+1. It doesn't guarantee that the first requester will be the first to get the lock.
+2. It doesn't guarantee access to the resource (a requester can wait indefenitely).
+
+To prevent [starvation](https://en.wikipedia.org/wiki/Starvation_%28computer_science%29), we add a simple FIFO queue 
+implemented in Redis, managing access to the resource. This queue allows us to guarantee access to a resource and guarantee
+ that the lock will be executed in order of lock request.
```
Comment (idjaw) [inline README.md]: remove the second guarantee.

**Comment by idjaw** [comment]
:+1:

**Pull request merged**